src/aiida/orm/authinfos.py

```diff
@@ -45,51 +48,60 @@ class AuthInfo(entities.Entity['BackendAuthInfo', AuthInfoCollection]):
     """ORM class that models the authorization information that allows a `User` to connect to a `Computer`."""
 
     _CLS_COLLECTION = AuthInfoCollection
+    PROPERTY_WORKDIR = 'workdir'
 
-    __qb_fields__ = [
-        add_field(
-            'enabled',
-            dtype=bool,
+    class Model(entities.Entity.Model):
+        computer: int = MetadataField(
+            description='The PK of the computer',
             is_attribute=False,
-            doc='Whether the instance is enabled',
-        ),
-        add_field(
-            'auth_params',
-            dtype=Dict[str, Any],
+            orm_class=Computer,
+            orm_to_model=lambda auth_info: auth_info.computer.pk,  # type: ignore[attr-defined]
+        )
+        user: int = MetadataField(
+            description='The PK of the user',
```

Comment:
Repeating a comment from `comments.py`, but again, `user` here is the PK, but in the `__init__`, `user` comes in as `User` and when PK is needed -> `user.pk`. What was your reasoning for dropping `_pk` in the models?

Reply:
This is another generic "problem" that arises due to the discrepancies in field types when comparing an ORM instance and its equivalent pydantic model. The Python ORM deals with instances of the ORM, i.e. an `AuthInfo` takes a `User` instance, even though in the database layer the link is represented by the pk of the `User` instance. Similarly, when serializing an `AuthInfo` to JSON, the `User` field also needs to be serialized, which we do by returning its pk as an integer.

From the ORM's perspective, it makes more sense to work with ORM instances, i.e. it is easier for the user (I would argue) for the constructor to request a `User` instance for the `user` argument as opposed to its pk. Same goes for other ORM classes like `Computer` etc. But the `pydantic` model is _outward_ facing where the `User` class makes no sense. So the `pydantic` model instead defines the `user` field as its serialized data type, i.e. an `int`. When a user will create a new `AuthInfo` over a REST API, they will be specifying its `user` by its pk and not through a `User` instance.

This discrepancy between the types of the ORM class constructor and its pydantic model will pop up in various places.  That is why these `orm_class` and `orm_to_model` attributes are there to allow the (de)serialization code in `Entity` to convert between the two. We _could_ choose to make it more consistent by changing the constructors to not take ORM class instances but always work with their pks. However this carries the downsides:

* it makes the ORM less intuitive to work with (IMHO)
* it is backwards incompatible

For the last point, it is perhaps possible to provide a deprecation pathway but I think this will still introduce a lot of hassle for users. That is why I opted for the current solution and keep the slight discrepancy between pydantic model and the ORM class constructor.

Reply:
Totally understand, though I was more commenting on the model key names - `user` instead of `user_pk` - and not on the types - `int` vs. `User`, the latter I understand from your comment 👍

Reply:
> I was more commenting on the model key names - user instead of user_pk - and not on the types - int vs. User

I see. The reason I didn't name it `user_pk` is because the serializer on the pydantic model will already convert it to an actual user. So if you do `pydantic_comment = Comment.Model(user=1)` then `assert isinstance(pydantic_comment.user, User)` will pass. In other words, the `user` field on a pydantic model instance will return a `User` instance and not the integer, but an integer is required to construct the model instance.

Reply:
I see. Thanks for clarifying 🙂

Reply:
If you rename the field to `user_pk` then there won't be a `user` attribute on the model instance. You would have `pydantic_comment.user_pk == User` which I find to be counterintuitive.

Reply:
Right, I understand that now after having played with the `Model` class a bit. All good 👍